src/taco-utils/telemetry.ts

```diff
@@ -379,15 +388,41 @@ module TacoUtility {
                 fs.writeFileSync(TelemetryUtils.telemetrySettingsFile, JSON.stringify(TelemetryUtils.telemetrySettings));
             }
 
-            private static getMachineId(): string {
+            private static getUniqueId(regValue: string, regHive: winreg.Hive, fallback: () => string): Q.Promise<any> {
+                var uniqueId: string;
+                var deferred: Q.Deferred<string> = Q.defer<string>();
+                if (os.platform() === "win32") {
+                    return TelemetryUtils.getRegistryValue(TelemetryUtils.REGISTRY_SQMCLIENT_NODE, regValue, regHive)
+                    .then(function(id: string): Q.Promise<string> {
+                        if (id) {
+                            uniqueId = id.replace(/[{}]/g, "");
+                            return Q.resolve(uniqueId);
+                        } else {
+                            return Q.resolve(fallback());
+                        }
+                    });
+                } else {
+                    return Q.resolve(fallback());
+                }
+            }
+
+            private static generateMachineId(): string {
+                var macAddress: string = TelemetryUtils.getMacAddress();
+                return crypto.createHash("sha256").update(macAddress, "utf8").digest("hex");
+            }
+
+            private static getMachineId(): Q.Promise<string> {
                 var machineId: string = TelemetryUtils.telemetrySettings.machineId;
                 if (!machineId) {
-                    var macAddress: string = TelemetryUtils.getMacAddress();
-                    machineId = crypto.createHash("sha256").update(macAddress, "utf8").digest("hex");
+                    return TelemetryUtils.getUniqueId(TelemetryUtils.REGISTRY_MACHINEID_VALUE, winreg.HKLM, TelemetryUtils.generateMachineId)
+                    .then(function(id: string): Q.Promise<string> {      
+                        TelemetryUtils.telemetrySettings.machineId = id;
```

Comment:
why are we setting all properties twice? once TelemetryUtils.machineId = id and also TelemetryUtils.telemetrySettings.machineId